bot/exts/recruitment/talentpool/_review.py

```diff
@@ -120,6 +127,65 @@ async def make_review(self, user_id: int) -> typing.Tuple[str, Optional[Emoji]]:
         review = "\n\n".join((opening, current_nominations, review_body, vote_request))
         return review, seen_emoji
 
+    async def archive_vote(self, message: PartialMessage, passed: bool) -> None:
+        """Archive this vote to #nomination-archive."""
+        message = await message.fetch()
+
+        # We consider that if a message has been sent less than 2 second before the one being archived
+        # it is part of the same nomination.
+        # For that we try to get messages sent in this timeframe until none is returned and NoMoreItems is raised.
+        messages = [message]
+        with contextlib.suppress(NoMoreItems):
+            while True:
+                new_message = await message.channel.history(  # noqa: B305 - yes flake8, .next() is a thing here.
+                    before=messages[-1].created_at,
+                    after=messages[-1].created_at - timedelta(seconds=2)
+                ).next()
+                messages.append(new_message)
```

Comment:
I'm really not sure about this section. If you know that you're holding the last message, why not iterate over the messages before it until the API call limit and breaking upon an unrelated message?

Reply:
I actually think it's safe in that channel to just not have a limit.

Reply:
Changed in https://github.com/python-discord/bot/pull/1549/commits/c111dd7db591450d62dfa590b11f2fc7078a16e3